src/opencl/opencl_utils.hpp

```diff
@@ -0,0 +1,242 @@
+#define CL_TARGET_OPENCL_VERSION 220
+
+#include <string>
+#include <vector>
+#include <fstream>
+#include <sstream>
+#include <cstring>
+#include <iostream>
+
+#ifdef __APPLE__
+#include <Open/opencl.h>

```

Comment:
Shouldn't this be:
`#include <OpenCL/cl.h>`
The original version did not work on my MacBook with intel CPU. Had to make this change manually.